src/wagtail_vector_index/index/base.py

```diff
@@ -9,16 +10,19 @@
 from wagtail_vector_index.backends import get_vector_backend
 
 
-@dataclass
+@dataclass(kw_only=True, frozen=True)
 class Document:
     """Representation of some content that is passed to vector storage backends.
 
     A document is usually a part of a model, e.g. some content split out from
     a VectorIndexedMixin model. One model instance may have multiple documents.
+
+    The embedding_pk on a Document must be the PK of an Embedding model instance.
     """
 
-    id: str
+    id: UUID
```

Comment:
**Question:**
- I'm wondering, do we use document ID for anything at all?
- Would this id be ever saved anywhere, so when the same document is loaded again, it has the same ID?

Reply:
We use it in the Qdrant backend, but we could just as well use the Embedding PK there which is likely to be more useful (and less likely to change), so you're right. I'll scrap ID for now.

Reply:
Done

Reply:
In some ways, it feels like we should axe any notion of a document ID in the codebase and just use `embedding_pk` explicitly everywhere.

Reply:
I think this is what this change does (except in some cases where we get an ID back from the vector backend)?

Reply:
> I think this is what this change does (except in some cases where we get an ID back from the vector backend)?

I guess our local Document object never has an actual ID, which this PR solidifies. Document ID in a remote backend is not the same as ID in our system I guess and the local Document object does not need to care where its source is, and if it did, it would likely need a different field than ID I guess.